src/lib.rs

```diff
@@ -110,53 +194,59 @@ fn random_gen_dust_concentration() -> String {
     rng.gen_range(0..=1000).to_string()
 }
 
-pub fn run(cli: &Cli) {
-    println!("\r\ncli: {:?}\r\n", cli);
-
+fn send_request(url: &str, client: Client) -> reqwest::Result<Response> {
     let dust_concentration = random_gen_dust_concentration();
     let air_purity = AirPurity::from_value(dust_concentration.parse::<u16>().unwrap()).to_string();
     let reading = Reading::new(
         random_gen_temperature(),
         random_gen_humidity(),
         random_gen_pressure(),
         dust_concentration,
-        air_purity
+        air_purity,
     );
 
     let json = serde_json::to_string(&reading).unwrap();
-    const URL: &str = "http://localhost:4000/api/readings/add";
-
-    println!("Sending POST request to {} as JSON: {}", URL, json);
-
-    let client = Client::new();
-    let res = client
-        .post(URL)
+    println!("Sending POST request to {} as JSON: {}", url, json);
+    client
+        .post(url)
         .header(CONTENT_TYPE, "application/json")
         .body(json)
-        .send();
-    match res {
-        Ok(response) => {
-            match cli.debug {
-                true => println!("Response from Ambi backend: {:#?}", response),
-                false => println!("Response from Ambi backend: {:?}", response.status().as_str())
+        .send()
+}
+
+pub fn run(cli: &Cli) {
+    println!("\r\ncli: {:?}\r\n", cli);
+
+    const URL: &str = "http://localhost:4000/api/readings/add";
+    let mut handlers: Vec<JoinHandle<()>> = vec![];
+    let debug: bool = cli.debug;
+    for _ in 0..cli.int {
+        let handler = spawn(move || match send_request(URL, Client::new()) {
+            Ok(response) => {
+                let result = RequestResult::new(
+                    String::from("Response from Ambi backend."),
+                    None,
+                    Some(response),
+                    std::thread::current().id(),
+                );
+                ResponseLog::new(debug, &mut std::io::stdout()).print(result)
```

Comment:
Whether the use of `new()` remains or you decide to take my advice and switch to a static instance of the logger, I think it makes ergonomic sense to separate instantiation of `ResponseLog` from using it `print()`. And then you can only create a single instance of it one time and not have to repeat instantiating it.